File: frontend/src/app/models/post-dto.ts
import { Comment } from './comment';
import { Like } from './like';
import { Post } from './post';

export interface PostDTO {
  post: Post;
  likes: Like[];
  comments: Comment[];
  showOptions?: boolean;
}
// Extend your PostDTO safely in the component file
export interface PostWithOptions extends PostDTO {
  showOptions?: boolean;
}

File: jee-liberty-posts-war/src/main/java/dto/PostDTO.java
package dto;

import java.util.List;

import model.Comment;
import model.Like;
import model.Post;

public class PostDTO {
    private Post post;
    private List<Like> likes;
    private List<Comment> comments;

    public List<Like> getLikes() {
        return likes;
    }

    public void setLikes(List<Like> likes) {
        this.likes = likes;
    }

    public List<Comment> getComments() {
        return comments;
    }

    public void setComments(List<Comment> comments) {
        this.comments = comments;
    }

    public Post getPost() {
        return post;

    }

    public PostDTO(Post post, List<Like> likes, List<Comment> comments) {
        this.post = post;
        this.likes = likes;
        this.comments = comments;
    }
}